leshan-client-core/src/main/java/org/eclipse/leshan/client/servers/RegistrationEngine.java

```diff
@@ -383,6 +383,12 @@ public void destroy(boolean deregister) {
         }
     }
 
+    public void forceUpdate() {
+        cancelUpdateTask(false);
+        LOG.info("Forcing update of registration...");
+        updateFuture = schedExecutor.schedule(new UpdateRegistrationTask(), 100, TimeUnit.MILLISECONDS);
```

Comment:
why using a delay ?

Reply:
The reasoning is that this is triggered by a server request, and a delay would decrease the risk of interfering with the response to that request. I'm no UDP expert, but a short delay seems less error prone. Maybe it should even be longer?

Reply:
It seems pretty binded to your specific use case. I will go for a scheduler.submit() and so you could add the delay you want on your side (this will be more flexible for you)

Reply:
Please clarify your last comment, I'm not sure what you mean.

Reply:
Replace schedExecutor.schedule() by schedExecutor.submit()

Reply:
OK